lsusb.c

```diff
@@ -3460,7 +3460,7 @@ static void dump_bos_descriptor(libusb_device_handle *fd, bool* has_ssp)
 	else if (bos_desc_static[0] != 5 || bos_desc_static[1] != USB_DT_BOS)
 		return;
 
-	bos_desc_size = bos_desc_static[2] + (bos_desc_static[3] << 8);
+	bos_desc_size = convert_le_u16(&bos_desc_static[2]);
```

Comment:
This is doing the same thing as the original code, right?  Why make this change?

Reply:
How is `+` incorrect here?  I agree that using the `convert_le_u16` function is more obvious and "cleaner",
but there is nothing wrong with the original code here that I can see, the end-result should be the same,
or am I missing something

Reply:
We should be doing OR operation not PLUS(addition).  Thus original code prints wrong value..

+static unsigned int convert_le_u16 (const unsigned char *buf)
+{
+	return buf[0] | (buf[1] << 8);
+}

Reply:
`|` is identical to `+` for this logic as far as I can tell because there will never be any overlapping bits, again, what am I missing?